data/pigui/modules/station-view/07-police.lua

```diff
@@ -1,17 +1,157 @@
 -- Copyright © 2008-2020 Pioneer Developers. See AUTHORS.txt for details
 -- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
 
+local Game = import "Game"
+local Rand = import "Rand"
 local ui = import 'pigui/pigui.lua'
 local StationView = import 'pigui/views/station-view'
 local Lang = import 'Lang'
-
+local Legal = import "Legal"
+local utils = import "utils"
+local InfoFace = import 'ui/PiguiFace'
+local Format = import "Format"
+local Character = import "Character"
 local l = Lang.GetResource("ui-core")
 
+local ModalWindow = import 'pigui/libs/modal-win.lua'
+
+local rescaleVector = ui.rescaleUI(Vector2(1, 1), Vector2(1600, 900), true)
+
+local pionillium = ui.fonts.pionillium
+local orbiteer = ui.fonts.orbiteer
+local colors = ui.theme.colors
+local face = nil
+local stationSeed = false
+
+local popup = ModalWindow.New('policePopup', function()
```

Comment:
> When trying to pay fines higher than what the player has, the modal popup window saying "not enough money" crashes the game when clicking "OK". I'm not sure what the correct way of implementing this is. Pointers are most welcome, ping @vakhoir

Sorry, this is my fault, I must have tested the modal window changes with a view where a window  reference was used rather than `self:close()`, this can be fixed by just adding `self` to the handler parameters. I'll go over the other views to fix this.

```suggestion
local popup = ModalWindow.New('policePopup', function(self)
```

Reply:
thanks, that fixed it. The missing `self` can be found in Lobby (and maybe others as well)